### PR TITLE
Add collection and generics lab

### PR DESCRIPTION
**Pull request by ialidzhikov**

### CODE REVIEW

File: 04-collections-and-generics/lab/01-car-store/README.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+public SportCar(Model model, int year, int price, EngineType engineType, Region region);
+```
+и да се намират в пакет `bg.sofia.uni.fmi.mjt.carstore.car`
+
+- `Model` може да приема следните стойности: **ALFA**, **AUDI**, **BMW**, **MERCEDES**, **FERRARI**, **OPEL**
+- `EngineType` може да приема следнтие стойности: **DIESEL**, **GASOLINE**, **ELECTIC**, **HYBRID**
+- `Region` може да приема следните стойности: **SOFIA**, **BURGAS**, **VARNA**, **PLOVDIV**, **RUSE**, **GABROVO**, **VIDIN**, **VRATSA**
+- Всяка кола се идентифицира уникално чрез нейния регистрационен номер. Той се образува по следния начин:
+``{region}{number}{random alphabet char}{random alphabet char}``, където **{number}** е число, започващо от 1000 и инкрементиращо се на всяка новосъздадена кола, **{random alphabet char}** e произволна буква от латинската азбука а **{region}** е един от следните областни кодове (буквите на кодовете са от латинската азбука):
```
Comment (100yo) [inline 04-collections-and-generics/lab/01-car-store/README.md]:
```suggestion
``{region}{number}{random alphabet char}{random alphabet char}``, където **{number}** е число, започващо от 1000 и инкрементиращо се при всяка новосъздадена кола, **{random alphabet char}** e произволна буква от латинската азбука а **{region}** е един от следните областни кодове (буквите на кодовете са от латинската азбука):
```

File: 04-collections-and-generics/lab/01-car-store/README.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+  /**
+  * Returns the unique registration number of the car.
+  */
+  public String getRegistrationNumber();
+
+}
+```
+
+Конкретните имплементации на `Car` са `OrdinaryCar` и `SportCar` и трябва да имат следните конструктори:
```
Comment (100yo) [inline 04-collections-and-generics/lab/01-car-store/README.md]:
```suggestion
Конкретните имплементации на `Car` са `OrdinaryCar` и `SportsCar` и трябва да имат следните конструктори:
```

File: 04-collections-and-generics/lab/01-car-store/README.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+```java
+public OrdinaryCar(Model model, int year, int price, EngineType engineType, Region region);
+
+
+public SportCar(Model model, int year, int price, EngineType engineType, Region region);
```
Comment (100yo) [inline 04-collections-and-generics/lab/01-car-store/README.md]:
```suggestion
public SportsCar(Model model, int year, int price, EngineType engineType, Region region);
```

**Comment by ialidzhikov** [comment]
Thanks to @yordanovagabriela , @KristianZH and @desislavaa .

**Pull request merged**